client/src/components/NewsFeed.js

```diff
@@ -0,0 +1,105 @@
+import React, { Component } from 'react';
+import { Segment, Feed, Image, Header, Icon, Divider, Modal, Grid} from 'semantic-ui-react'
+import { makeCall } from '../apis';
+import AlumniProfile from './AlumniProfile'
+
+/*
+ * DETAILS:
+ * NewsFeed component for visual display of news items created on backend
+ * Each type of post will recieve its own create(EVENT TYPE)Post method,
+ * called by constructDisplay
+ * PROPS
+ * userDetails - full profile of logged in user
+ * role - used to determine which types of news will be pulled
+ */
+export default class NewsFeed extends Component {
+    state={
+        newsItems: [],
+        display: []
+    }
+
+    async componentWillMount() {
+        let newsItems = await this.getNews(this.props.userRole)
+        await this.setState({
+            newsItems: newsItems.news.reverse()
+        })
+        this.constructDisplay()
+    }
+
+    getNews(role) {
+        return makeCall({}, '/news/getNews/'+{role}, 'get')
+    }
+
+    constructDisplay() {
+        let display = []
+        let i = 0
+        for (let feedItem of this.state.newsItems) {
+            if (feedItem.event === 'New Alumni') {
+                display.push(<Divider key={i}/>)
+                display.push(this.createNewAlumniPost(feedItem))
+            }
+            i++;
+        }
+        this.setState({display: display})
+    }
+
+    /*
+     * EVENT: 'New Alumni'
+     * Display to: 'BOTH'
+     * Contains: Alumni[0] and time (stored as string) ONLY
+     * On Click: Opens a profile modal that can not be interacted with
+     */
+    createNewAlumniPost(feedItem) {
+        let alumniDetails = feedItem.alumni[0];
+        let elapsedTime = new Date();
+        let uploadedTime = new Date(feedItem.time);
```

Comment:
Let's make a TODO to move this calculation to the backend, at the time we use end-point to get feedItems.
We can use [momentjs](https://stackoverflow.com/questions/42186717/moment-js-show-diff-between-2-dates)